pcbdraw/pcbdraw.py

```diff
@@ -636,10 +659,20 @@ def load_remapping(remap_file):
 
 def adjust_lib_path(path):
     if path == "default" or path == "kicad-default":
-        return os.path.join(PKG_BASE, "footprints", "KiCAD-base")
+        return [os.path.join(p, FOOTPRINTS_SUBDIR, "KiCAD-base") for p in data_path]
     if path == "eagle-default":
-        return os.path.join(PKG_BASE, "footprints", "Eagle-export")
-    return path
+        return [os.path.join(p, FOOTPRINTS_SUBDIR, "Eagle-export") for p in data_path]
+    return [path]
+
+def setup_data_path():
+    global data_path
+    share = os.path.join('share', 'pcbdraw')
+    if os.name == 'posix':
+        data_path.append(os.path.join(sysconfig.get_path('data', 'posix_user'), share))
+        data_path.append(os.path.join(sysconfig.get_path('data', 'posix_prefix'), share))
+    elif os.name == 'nt':
+        data_path.append(os.path.join(sysconfig.get_path('data', 'nt_user'), share))
+        data_path.append(os.path.join(sysconfig.get_path('data', 'nt'), share))
```

Comment:
I would do

```python
try:
    if os.name == 'posix':
        data_path.append(os.path.join(sysconfig.get_path('data', 'posix_user'), share))
        data_path.append(os.path.join(sysconfig.get_path('data', 'posix_prefix'), share))
    elif os.name == 'nt':
        data_path.append(os.path.join(sysconfig.get_path('data', 'nt_user'), share))
        data_path.append(os.path.join(sysconfig.get_path('data', 'nt'), share))
    else:
        data_path.append(os.path.join(sysconfig.get_path('data'), share))
except KeyError:
    data_path.append(os.path.join(sysconfig.get_path('data'), share))
```

This makes sure everything will still work on other custom platforms, or more exotic interpreter implementations. This will use the default scheme in case anything fails or if we are on an unknown platform, the only drawback is that it won't support the user directory.

Reply:
It looks ok. The only detail is that KeyError is documented for get_paths, not for get_path. I tried and it raises KeyError for bogus values.
I coded it in a way that follows the docs:

*If scheme is provided, it must be a value from the list returned by get_scheme_names(). Otherwise, the default scheme for the current platform is used.*

So I'm calling get_scheme_names() and checking if the scheme is valid.